CONTRIBUTING.md

```diff
@@ -0,0 +1,254 @@
+happy to accept your patches.
+
+### Documentation
+
+First of all, all new features and all user-visible changes to existing
+features need to be documented. That means editing the appropriate help files
+and adding a NEWS entry where appropriate. When editing the NEWS file, you
+should add your change to the top of the list of changes. If your change alters
+an interface (key binding, command) and is likely to cause trouble, prefix it
+with `IMPORTANT:`, otherwise, place it below the other `IMPORTANT` entries. If
+you're not sure if your change merits a news entry, or if it's important,
+please ask.
+
+### Coding Style
+
+In general: Just look at the existing source code!
+
+We try to target experienced JavaScript developers who do not necessarily need
+to have a good understanding of Vimperator's source code, nor necessarily
+understand in-depth concepts of other languages like Lisp or Python. Therefore,
+the coding style should feel natural to any JavaScript developer. Of course,
+this does not mean, you have to avoid all new JavaScript features like list
+comprehension or generators. Use them, when they make sense, but don't use them
+when the resulting code is hard to read.
+
+#### The most important style issues are:
+
+- Use 4 spaces to indent things, no tabs, not 2, nor 8 spaces. If you use Vim,
+  this should be taken care of automatically by the modeline (like the one
+  below).
+
+- No trailing whitespace.
+
+- Use `"` for enclosing strings instead of `'`, unless using `'` avoids
+  escaping of lots of `"`:
+
+  ```javascript
+  alert("foo")
+
+  alert('foo')
+  ```
+
+- Use `//` regexp literals rather than RegExp constructors, unless you're
+  constructing an expression on the fly, or RegExp constructors allow you to
+  escape less `/s` than the additional escaping of special characters required
+  by string quoting:
+
+  ```javascript
+  // Good
+  /application\/xhtml\+xml/
+  RegExp("http://(www\\.)vimperator.org/(.*)/(.*)")
+
+  // Bad
+  RegExp("application/xhtml\\+xml")
+  /http:\/\/(www\.)vimperator.org\/(.*)\/(.*)/
+  ```
+
+- Exactly one space after `if/for/while/catch` etc. and after a comma, but none
+  after a parenthesis or after a function call:
+
+  ```javascript
+  for (pre; condition; post)
+  alert("foo");
+  ```
+
+- Bracing is formatted as follows:
+
+  ```javascript
+  function myFunction () {
+      if (foo)
+          return bar;
+      else {
+          baz = false;
+          return baz;
+      }
+  }
+  var quux = frob("you",
+      {
+          a: 1,
+          b: 42,
+          c: {
+              hoopy: "frood"
+          }
+      });
+  ```
+
+  When in doubt, look for similar code.
+
+- No braces for one-line conditional statements:
+
+  ```javascript
+  if (foo)
+      frob();
+  else
+      unfrob();
+  ```
+
+- Prefer lambda-style functions where suitable:
+
+  ```javascript
+  // Good
+  list.filter(function (elem) elem.good != elem.BAD);
```

Comment:
this is no longer valid. "Expression closures" are being removed plus non-standard.
`function (elem) returnValue` that is.